### ISSUE TITLE
DataType Examples are not parsed

### ISSUE THREAD

**Issue by comikem** [mulesoft/api-workbench#297]
In attempting to use dataTypes in place of schemas I have defined a type with an example.  The example in the type is ignored by the console or mocking service unless the example is explicitly defined in the response's specific media type.  

I raised this issue on the api-designer repo as well, and it was confirmed a bug this morning.

Example:
Type Definition:
```RAML
types:
  Member:
    properties:
      firstName: string
      lastName: string
      age: integer
    examples:
      Bob:
        value:
          firstName: "Bob"
          lastName: "Slidell"
          age: 42
      Bill:
        value:
          firstName: "Bill"
          lastName: "Lumberg"
          age: 41
````

If a response is resource type the example is not included
````RAML
resourceTypes:
  collection:
    usage: Use this resourceType to represent a collection of <<resourcePathName|!singularize>> items
    description: A collection of <<resourcePathName>>
    get:
      description: |
        Get all <<resourcePathName>>,
        optionally filtered
      responses:
        200:
          body:
            type: <<typeName|!singularize>>[]
````
Resource:
````RAML
/members:
  type:
     collection:
       typeName: <<resourcePathName|!uppercamelcase>>
  get:
````
This will show the Object type and its properties but not the example.  If the resource type explicitly defines the example it will be included in the console and returned by the Mock service.

````RAML
resourceTypes:
  collection:
    usage: Use this resourceType to represent a collection of <<resourcePathName|!singularize>> items
    description: A collection of <<resourcePathName>>
    get:
      description: |
        Get all <<resourcePathName>>,
        optionally filtered
      responses:
        200:
          body:
            type: <<typeName|!singularize>>[]
            application/json:            
              examples:
                Bob:
                  value:
                    firstName: "Bob"
                    lastName: "Slidell"
                    age: 42
                Bill:
                  value:
                    firstName: "Bill"
                    lastName: "Lumberg"
                    age: 41
````

The description of Examples in the 1.0 spec seems to indicate this would be a supported usage.  Full RAML below:
````RAML
#%RAML 1.0
title: TestRAML
baseUri: http://someapi.com/api
mediaType: application/json

resourceTypes:
  collection:
    usage: Use this resourceType to represent a collection of <<resourcePathName|!singularize>> items
    description: A collection of <<resourcePathName>>
    get:
      description: |
        Get all <<resourcePathName>>,
        optionally filtered
      responses:
        200:
          body:
            type: <<typeName|!singularize>>[]

types:
  Member:
    properties:
      firstName: string
      lastName: string
      age: integer
    examples:
      Bob:
        value:
          firstName: "Bob"
          lastName: "Slidell"
          age: 42
      Bill:
        value:
          firstName: "Bill"
          lastName: "Lumberg"
          age: 41
/members:
  type:
     collection:
       typeName: <<resourcePathName|!uppercamelcase>>
  get:
````

**Comment by ddenisenko** [comment]
Could you please let me know, which behavior you expect from API Workbench in this case?

If you meant that this is a parser bug and accidentally posted here instead of the parser repository https://github.com/raml-org/raml-js-parser-2/issues , you can repost it there, but the parser output looks fine for me:

```
{
  "ramlVersion": "RAML10",
  "type": "Api",
  "specification": {
    "types": [
      {
        "Member": {
          "name": "Member",
          "displayName": "Member",
          "type": [
            "object"
          ],
          "examples": [
            {
              "value": "{\n  \"firstName\": \"Bob\",\n  \"lastName\": \"Slidell\",\n  \"age\": 42\n}",
              "strict": true,
              "name": "Bob",
              "structuredValue": {
                "firstName": "Bob",
                "lastName": "Slidell",
                "age": 42
              }
            },
            {
              "value": "{\n  \"firstName\": \"Bill\",\n  \"lastName\": \"Lumberg\",\n  \"age\": 41\n}",
              "strict": true,
              "name": "Bill",
              "structuredValue": {
                "firstName": "Bill",
                "lastName": "Lumberg",
                "age": 41
              }
            }
          ],
          "required": true,
          "properties": {
            "firstName": {
              "name": "firstName",
              "displayName": "firstName",
              "type": [
                "string"
              ],
              "required": true,
              "__METADATA__": {
                "primitiveValuesMeta": {
                  "displayName": {
                    "calculated": true
                  },
                  "required": {
                    "insertedAsDefault": true
                  }
                }
              }
            },
            "lastName": {
              "name": "lastName",
              "displayName": "lastName",
              "type": [
                "string"
              ],
              "required": true,
              "__METADATA__": {
                "primitiveValuesMeta": {
                  "displayName": {
                    "calculated": true
                  },
                  "required": {
                    "insertedAsDefault": true
                  }
                }
              }
            },
            "age": {
              "name": "age",
              "displayName": "age",
              "type": [
                "integer"
              ],
              "required": true,
              "__METADATA__": {
                "primitiveValuesMeta": {
                  "displayName": {
                    "calculated": true
                  },
                  "required": {
                    "insertedAsDefault": true
                  }
                }
              }
            }
          },
          "__METADATA__": {
            "primitiveValuesMeta": {
              "displayName": {
                "calculated": true
              },
              "type": {
                "insertedAsDefault": true
              },
              "required": {
                "insertedAsDefault": true
              }
            }
          }
        }
      }
    ],
    "resourceTypes": [
      {
        "collection": {
          "description": "A collection of <<resourcePathName>>",
          "name": "collection",
          "usage": "Use this resourceType to represent a collection of <<resourcePathName|!singularize>> items",
          "get": {
            "responses": {
              "200": {
                "code": "200",
                "body": {
                  "body": {
                    "name": "body",
                    "displayName": "body",
                    "type": [
                      "<<typeName|!singularize>>[]"
                    ],
                    "required": true,
                    "__METADATA__": {
                      "primitiveValuesMeta": {
                        "displayName": {
                          "calculated": true
                        },
                        "required": {
                          "insertedAsDefault": true
                        }
                      }
                    }
                  }
                }
              }
            },
            "description": "Get all <<resourcePathName>>,\noptionally filtered\n",
            "method": "get"
          }
        }
      }
    ],
    "title": "TestRAML",
    "baseUri": "http://someapi.com/api",
    "protocols": [
      "HTTP"
    ],
    "mediaType": "application/json",
    "resources": [
      {
        "methods": [
          {
            "responses": {
              "200": {
                "code": "200",
                "body": {
                  "application/json": {
                    "name": "application/json",
                    "displayName": "application/json",
                    "type": [
                      "Member[]"
                    ],
                    "required": true,
                    "__METADATA__": {
                      "primitiveValuesMeta": {
                        "displayName": {
                          "calculated": true
                        },
                        "required": {
                          "insertedAsDefault": true
                        }
                      }
                    }
                  }
                }
              }
            },
            "protocols": [
              "HTTP"
            ],
            "description": "Get all members,\noptionally filtered\n",
            "method": "get",
            "__METADATA__": {
              "primitiveValuesMeta": {
                "protocols": {
                  "calculated": true
                }
              }
            }
          }
        ],
        "type": {
          "collection": {
            "typeName": "<<resourcePathName|!uppercamelcase>>"
          }
        },
        "description": "A collection of members",
        "relativeUri": "/members",
        "displayName": "/members",
        "__METADATA__": {
          "primitiveValuesMeta": {
            "displayName": {
              "calculated": true
            }
          }
        },
        "relativeUriPathSegments": [
          "members"
        ]
      }
    ],
    "__METADATA__": {
      "primitiveValuesMeta": {
        "protocols": {
          "calculated": true
        }
      }
    }
  },
  "errors": []
}
```

As you can see, there is `Member[]` type in `get` method, and `Member` type in `types` does list both of your examples in two different forms.

**Comment by sichvoge** [comment]
I think that shouldn't actually be here! Its an issue for the mocking service / API Designer only.

**Comment by comikem** [comment]
Sorry for the confusion.  I was seeing a difference in behavior in the api-workbench console which looks similar to the issue with API Designer.  If the console portion of the package is another project or if I've got an error in the spec please let me know.  Thanks for looking into this!
If the example is part of the type, the console only displays the type meta-data.
![screen shot 2016-08-18 at 7 40 08 am](https://cloud.githubusercontent.com/assets/18690628/17776003/0ffd4d5e-6518-11e6-8a25-a86711b8369d.png)

If the example is explicitly set the console will show the example and type instead.
![screen shot 2016-08-18 at 7 54 30 am](https://cloud.githubusercontent.com/assets/18690628/17776286/11cfe06e-6519-11e6-8678-8ec4939d6500.png)

**Comment by ddenisenko** [comment]
I agree, the samples defined as on a screenshot above should appear in the console.

**Comment by ddenisenko** [comment]
Fixed. Fix will be published in 0.8.39.
